### PR TITLE
add http caching on GitHub API calls when running danger local [#198]

### PR DESCRIPTION
**Pull request by dbgrandi**
- You can clear the cache with the `--clear-http-cache` command line arg.
- The cache TTL is 5 minutes

Still have 2 or 3 non-cached calls to GitHub, but it used to be around 8 when I was testing locally.

### CODE REVIEW

**Comment by dbgrandi** [comment]
(We should have a Danger rule that ensures a change to the `.gemspec` has a version restriction.)

**Comment by orta** [comment]
We could use `gem build` and look for warnings

**Comment by orta** [comment]
I'm doing something similar for YARD ATM

**Comment by orta** [comment]
To make that work you need this spec to pass:

``` ruby
      it "a multiline warning with markdown, no errors" do
        warnings = violations(["a markdown [link to danger](https://github.com/danger/danger)\n\n```\ncode\n```\n\nHello"])
        result = @g.generate_comment(warnings: warnings, errors: [], messages: [])
        # rubocop:disable Metrics/LineLength
        expect(result.gsub(/\s+/, "")).to eq(
          '<table><thead><tr><thwidth="50"></th><thwidth="100%"data-kind="Warning">1Warning</th></tr></thead><tbody><tr><td>:warning:</td><tddata-sticky="false"><p>amarkdown<ahref="https://github.com/danger/danger">linktodanger</a></p><p><code>code</code></p><p>Hello</p></td></tr></tbody></table><palign="right"data-meta="generated_by_danger">Generatedby:no_entry_sign:<ahref="http://danger.systems/">danger</a></p>'
        )
        # rubocop:enable Metrics/LineLength
      End
```

File: danger.gemspec [hunk 1]
```diff
@@ -25,6 +25,7 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "git", "~> 1"
   spec.add_runtime_dependency "colored", "~> 1.2"
   spec.add_runtime_dependency "faraday", "~> 0"
+  spec.add_runtime_dependency "faraday-http-cache", "~> 1.0"
```
Comment (KrauseFx) [inline danger.gemspec]: For our use case, wouldn't it be enough to just cache the GitHub JSON response ourself?

Comment (dbgrandi) [inline danger.gemspec]: I went this way to also get caching of GitHub API calls that are made from within the Dangerfile or plugins. Figured there is enough logic in how to properly cache that I didn't want to duplicate it.

**Comment by orta** [comment]
👍

**Comment by orta** [comment]
I've got the multi-line fix in #343

**Pull request merged**